src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/custom.py

```diff
@@ -45,6 +74,8 @@ def _build_cloud(cloud_name, cloud_config=None, cloud_args=None):
             setattr(c.endpoints, arg.replace('endpoint_', ''), cloud_args[arg])
         elif arg.startswith('suffix_') and cloud_args[arg] is not None:
             setattr(c.suffixes, arg.replace('suffix_', ''), cloud_args[arg])
+    arm_endpoint = cloud_args.get('endpoint_resource_manager', None)
+    _populate_from_metadata_endpoint(c, arm_endpoint)
```

Comment:
What happens the user provides endpoint_resource_manager and active_directory_resource_id params?  We override the user provided value for active_directory_resource_id  with the value from metadata all the time?